### PR TITLE
Added release status / latest vs stable information

### LINKED ISSUE #910: add release status page
document the state:

latest release : 0.5.0

stable release: 0.2.12

possibly mentioning https://archive.istio.io/v0.2/docs/
and
`curl -L https://git.io/getIstio` (pointing to 0.2.12)
vs
`curl -L https://git.io/getLatestIstio `

cc @costinm per slack chat, @hklai (about when we want 0.5.0 to be mentioned)

### PR DESCRIPTION
**Pull request by ldemailly**
Fixes #910

### CODE REVIEW

**Comment by ldemailly** [comment]
it looks like this:

screen shot 2018-01-31 at 2 40 23 pm

File: _data/istio.yml [hunk 1]
```diff
@@ -1 +1 @@
-version: 0.4.0
+version: 0.5.0
```
Comment (ldemailly) [inline _data/istio.yml]: this pr should go in with #902

**Comment by ldemailly** [comment]
@geeknoid PTAL

File: _about/notes/index.md [hunk 2]
```diff
@@ -1,7 +1,5 @@
 ---
 title: Release Notes
-overview: What's been happening with Istio.
```
Comment (geeknoid) [inline _about/notes/index.md]: Why remove the overview? That string is used for the hover text in the sidenav, and as the description test if you go to about/index.html

Comment (ldemailly) [reply _about/notes/index.md]: the text showed up on the page and looked strange with the new content - do you have a proposed new/better text or you want me to keep this one ?

Comment (ldemailly) [reply _about/notes/index.md]: updated. ptal

**Comment by ldemailly** [comment]
updated preview:
screen shot 2018-01-31 at 3 59 59 pm

**Comment by geeknoid** [comment]
Oh, I see. Things aren't quite right in general if you try to put actual
content on a section index page. It's fine to remove the overview then.

**Comment by ldemailly** [comment]
our story on stable vs not etc isn't super solid so I didn't want to create a new proeminent page yet I think we do need to document the 2 vanity getIstio*.io urls as well as what was discussed on slack with respect to baking release so this existing page seems like a good compromise: not too prominent yet discoverable

**Pull request merged**